Source/WebKit/webpushd/mac/com.apple.WebKit.webpushd.sb.in

```diff
@@ -22,8 +22,308 @@
+        "kern.maxfilesperproc"
+        "kern.memorystatus_level"
+        "kern.osproductversion" ;; <rdar://problem/51756739>
+        "kern.osrelease"
+        "kern.ostype"
+        "kern.osvariant_status"
+        "kern.osversion"
+        "kern.safeboot"
+        "kern.version"
+        "machdep.cpu.brand_string"
+        "security.mac.sandbox.sentinel"
+        "kern.tcsm_enable"
+        "kern.tcsm_available"
+        "vm.footprint_suspend")
+    (sysctl-name-prefix "net.routetable")
+    (sysctl-name-prefix "hw.optional.") ;; <rdar://problem/71462790>
+    (sysctl-name-prefix "hw.perflevel") ;; <rdar://problem/76783596>
+)
+
+(allow sysctl-write
+    (sysctl-name
+        "kern.tcsm_enable"))
+
+(deny mach-lookup (xpc-service-name-prefix ""))
+
+;; Utility functions for home directory relative path filters
+(define (home-regex home-relative-regex)
+  (regex (string-append "^" (regex-quote (param "HOME_DIR")) home-relative-regex)))
+
+(define (home-subpath home-relative-subpath)
+  (subpath (string-append (param "HOME_DIR") home-relative-subpath)))
+
+(define (home-literal home-relative-literal)
+  (literal (string-append (param "HOME_DIR") home-relative-literal)))
+
+(define (home-prefix home-relative-prefix)
+  (prefix (string-append (param "HOME_DIR") home-relative-prefix)))
+
+(define (allow-read-directory-and-issue-read-extensions path)
+    (if path
+        (begin
+            (allow file-read* (subpath path))
+            (allow file-issue-extension (require-all (extension-class "com.apple.app-sandbox.read") (subpath path))))))
+
+;; Read-only preferences and data
+(allow file-read*
+    ;; Basic system paths
+    (subpath "/Library/Frameworks")
+    (subpath "/Library/Managed Preferences")
+    (regex #"^/private/etc/(hosts|group|passwd)$")
+
+    ;; System and user preferences
+    (home-literal "/.CFUserTextEncoding"))
+
+(allow file-read-data
+    (literal "/usr/local/lib/log")) ; <rdar://problem/36629495>
+
+;; Preferences support
+(shared-preferences-read
+    "com.apple.LaunchServices"
+    "com.apple.crypto")
+
+(allow-reading-global-preferences)
+
+; FIXME: This is needed for some security framework calls (that use non-CFPreferences readers)
+(allow file-read-data
+    (literal "/Library/Preferences/com.apple.security.plist")
+    (home-subpath "/Library/Preferences/com.apple.security.plist"))
+
+;; On-disk WebKit2 framework location, to account for debug installations outside of /System/Library/Frameworks,
+;; and to allow issuing extensions.
+(allow-read-directory-and-issue-read-extensions (param "WEBKIT2_FRAMEWORK_DIR"))
+
+#if USE(SYSTEM_CONTENT_PATH)
+(map allow-read-directory-and-issue-read-extensions
+    read-directory-and-issue-read-extension-secondary-paths)
+#endif
+
+(allow mach-lookup
+#if PLATFORM(MAC)
+    (with report) (with telemetry)
+#endif
+#if !ENABLE(CFPREFS_DIRECT_MODE)
+    (global-name "com.apple.cfprefsd.agent")
+#endif
+    (global-name "com.apple.system.logger")
+)
```

Comment:
I think this can possibly be removed. ENABLE(CFPREFS_DIRECT_MODE) is 1 (also for webpushd, I believe) and the logger service is being blocked in other WebKit profiles (except for the root user, which may not apply in this case). Are you also calling _CFPrefsSetDirectModeEnabled and _CFPrefsSetReadOnly in webpushd? These can be called before the first CF prefs call, in order to avoid connection attempts to the CF prefs service.

Reply:
Ok, I'll change WebPushDaemon to enable direct mode.